src/raycast.cpp

```diff
@@ -17,11 +17,54 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#include "raycast.h"
 #include "irr_v3d.h"
 #include "irr_aabb3d.h"
+#include "constants.h"
+
+bool RaycastSort::operator()(const PointedThing &pt1,
+	const PointedThing &pt2) const
+{
+	// returns false if pt1 is nearer than pt2
+	if (pt1.distanceSq < pt2.distanceSq) {
+		return false;
+	}
+	if (pt1.distanceSq == pt2.distanceSq) {
+		// Sort them to allow only one order
+		if (pt1.type == POINTEDTHING_OBJECT) {
+			if (pt2.type == POINTEDTHING_OBJECT) {
```

Comment:
i think you can inline this condition as return (pt2.type == PT_OBJ && pt1.object_id < pt2.object_id);